2-classes-db/task.md

```diff
@@ -1,79 +1,140 @@
 
 You can work with it by running
+
 ```
 psql homework
 ```
+
 ## Submission
 
 Below you will find a set of tasks for you to complete to consolidate and extend your learning from this week. You will find it beneficial to complete the reading tasks before attempting some of these.
 
 To submit this homework write the correct commands after each question.
 
 ### 1. Show the date, transaction_no, description and amount for those transactions whose amount is between £30,000 and £31,000.
-```sql
 
+```sql
+SELECT date, transaction_no, description, amount
+FROM spends
+WHERE amount BETWEEN 30000 AND 31000;
 ```
+
 ### 2. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'fee'.
-```sql
 
+```sql
+SELECT date, transaction_no, supplier_inv_no, description, amount
+FROM spends
+WHERE description LIKE '%fee%';
 ```
+
+If we don't want to includes "fees" we will write as '%fee %'
+
 ### 3. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'Fee'.
-```sql
 
+```sql
+SELECT date, transaction_no, supplier_inv_no, description, amount
+FROM spends
+WHERE description LIKE '%Fee%';
 ```
+
+<!-- If we don't want to includes "Fees" we will write as '%fee %'  -->
+
 ### 4. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'fee' (case insensitive). You will need to search 'https://www.postgresql.org/docs/' to solve this.
-```sql
 
+```sql
+SELECT date, transaction_no, supplier_inv_no, description, amount
+FROM spends
+WHERE LOWER(description) LIKE '%fee%';
 ```
+
 ### 5. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose amount is £25,000, £30,000, £35,000 or £40,000.
-```sql
 
+```sql
+SELECT date, transaction_no, supplier_inv_no, description, amount
+FROM spends
+WHERE  amount in (25000, 30000, 35000, 40000);
 ```
+
 ### 6. Show the date, the supplier_id, the description and the amount for transactions with the expense area of 'Better Hospital Food'. You could do a query to get the expense_area_id first then do a query to find the dates, supplier_ids and amounts. But it would be better to do this all in one query by linking the tables together using INNER JOINs.
+
 ```sql
+SELECT date, supplier_id, description, amount
+FROM spends
+INNER JOIN expense_areas
+ON  expense_areas.expense_area='Better Hospital Food';
 
 ```
+
+<!-- SELECT date, supplier_id, description, amount,expense_areas.expense_area
+FROM spends
+INNER JOIN expense_areas
+ON  expense_areas.expense_area='Better Hospital Food'; -->
+
 ### 7. Show the date, supplier name, description and amount for transactions with the expense area of 'Better Hospital Food'. You will need to INNER JOIN another table to be able to do this.
-```sql
 
+```sql
+SELECT spends.date, suppliers.supplie, spends.description, spends.amount
```

Comment:
Same comment as above applies here, this JOIN isn't correct

Reply:
SELECT date, suppliers.supplier, description, amount
FROM spends
JOIN expense_areas ON spends.expense_area_id = expense_areas.id
JOIN suppliers ON spends.supplier_id = suppliers.id
WHERE expense_area = 'Better Hospital Food';